sdk/include/reactos/rosctrls.h

```diff
@@ -581,3 +581,254 @@ class CTreeView :
+    {
+        TTTOOLINFOW info = { sizeof(TTTOOLINFOW), 0 };
+        info.hwnd = hwndToolOwner;
+        info.uId = uId;
+        info.lpszText = pBuffer;
+        SendMessageW(TTM_GETTEXT, cchBuffer, reinterpret_cast<LPARAM>(&info));
+    }
+
+    VOID UpdateTipText(IN HWND hwndToolOwner, IN UINT uId, IN PCWSTR szText, IN HINSTANCE hinstResourceOwner = NULL)
+    {
+        TTTOOLINFOW info = { sizeof(TTTOOLINFOW), 0 };
+        info.hwnd = hwndToolOwner;
+        info.uId = uId;
+        info.lpszText = const_cast<PWSTR>(szText);
+        info.hinst = hinstResourceOwner;
+        SendMessageW(TTM_UPDATETIPTEXT, 0, reinterpret_cast<LPARAM>(&info));
+    }
+
+    BOOL EnumTools(IN CONST TTTOOLINFOW * pInfo)
+    {
+        return SendMessageW(TTM_ENUMTOOLS, 0, reinterpret_cast<LPARAM>(pInfo));
+    }
+
+    BOOL GetCurrentTool(OUT OPTIONAL TTTOOLINFOW * pInfo = NULL)
+    {
+        return SendMessageW(TTM_GETCURRENTTOOL, 0, reinterpret_cast<LPARAM>(pInfo));
+    }
+
+    VOID GetTitle(TTGETTITLE * pTitleInfo)
+    {
+        SendMessageW(TTM_GETTITLE, 0, reinterpret_cast<LPARAM>(pTitleInfo));
+    }
+
+    BOOL SetTitle(PCWSTR szTitleText, WPARAM icon = 0)
+    {
+        return SendMessageW(TTM_SETTITLE, icon, reinterpret_cast<LPARAM>(szTitleText));
+    }
+
+    VOID TrackActivate(IN HWND hwndToolOwner, IN UINT uId)
+    {
+        TTTOOLINFOW info = { sizeof(TTTOOLINFOW), 0 };
+        info.hwnd = hwndToolOwner;
+        info.uId = uId;
+        SendMessageW(TTM_TRACKACTIVATE, TRUE, reinterpret_cast<LPARAM>(&info));
+    }
+
+    VOID TrackDeactivate()
+    {
+        SendMessageW(TTM_TRACKACTIVATE, FALSE, NULL);
+    }
+
+    VOID TrackPosition(IN WORD x, IN WORD y)
+    {
+        SendMessageW(TTM_TRACKPOSITION, 0, MAKELPARAM(x, y));
+    }
+
+    // Opens the tooltip
+    VOID Popup()
+    {
+        SendMessageW(TTM_POPUP);
+    }
+
+    // Closes the tooltip - Pressing the [X] for a TTF_CLOSE balloon is equivalent to calling this
+    VOID Pop()
+    {
+        SendMessageW(TTM_POP);
+    }
+
+    // Delay times for AUTOMATIC tooltips (they don't affect balloons)
+    INT GetDelayTime(UINT which)
+    {
+        return SendMessageW(TTM_GETDELAYTIME, which);
+    }
+
+    VOID SetDelayTime(UINT which, WORD time)
+    {
+        SendMessageW(TTM_SETDELAYTIME, which, MAKELPARAM(time, 0));
+    }
+
+    // Activates or deactivates the automatic tooltip display when hovering a control
+    VOID Activate(IN BOOL bActivate = TRUE)
+    {
+        SendMessageW(TTM_ACTIVATE, bActivate);
+    }
+
+    // Adjusts the position of a tooltip when used to display trimmed text
+    VOID AdjustRect(IN BOOL bActivate = TRUE)
+    {
+        SendMessageW(TTM_ADJUSTRECT, bActivate);
+    }
+
+    // Useful for TTF_ABSOLUTE|TTF_TRACK tooltip positioning
+    SIZE GetBubbleSize(OUT TTTOOLINFOW * pInfo)
+    {
+        DWORD ret = SendMessageW(TTM_GETBUBBLESIZE, 0, reinterpret_cast<LPARAM>(pInfo));
```

Comment:
If it fails, maybe return 0,0 ?

Reply:
If it fails, the return value is FALSE, aka 0, so decoding it will result in {0,0}.

Reply:
right, i was not paying attention and assumed the size below was derived from the LPARAM